[Bug] startswith on a string does not work in a for loop

## Summary

I tried using the new fancy methods on a string.
I used the `startswith` method inside a for loop context and got a `nil` pointer.

### System specs

```go.mod
// arm64
module main

go 1.23.1

require github.com/nikolalohinski/gonja/v2 v2.3.1

require (
        github.com/dustin/go-humanize v1.0.1 // indirect
        github.com/json-iterator/go v1.1.12 // indirect
        github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
        github.com/modern-go/reflect2 v1.0.2 // indirect
        github.com/pkg/errors v0.9.1 // indirect
        github.com/sirupsen/logrus v1.9.3 // indirect
        golang.org/x/exp v0.0.0-20240404231335-c0f41cb1a7a0 // indirect
        golang.org/x/sys v0.19.0 // indirect
        golang.org/x/text v0.14.0 // indirect
)
```

## Example program

```golang
package main

import (
	"os"

	"github.com/nikolalohinski/gonja/v2"
	"github.com/nikolalohinski/gonja/v2/exec"
)

func main() {
	template, err := gonja.FromString(`
{% set ns = namespace(found=false) %}
{%- for v in ["a", "b", "c"] %}
      {%- if v.startswith("b") %}
        {% set ns.found=true %}
      {%- endif %}
{% endfor %}
{{ ns.found }}
{{ name }}
`)
	if err != nil {
		panic(err)
	}

	data := exec.NewContext(map[string]interface{}{
		"name": "bob",
	})

	if err = template.Execute(os.Stdout, data); err != nil { // Prints: Hello Bob!
		panic(err)
	}
}
```

## Error panic

```log
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1042100dc]

goroutine 1 [running]:
github.com/nikolalohinski/gonja/v2/exec.(*MethodSet[...]).Get(0x1c?, {0x1400015c116?, 0x2?})
        /Users/A108073420/go/pkg/mod/github.com/nikolalohinski/gonja/v2@v2.3.1/exec/environment.go:234 +0x2c
github.com/nikolalohinski/gonja/v2/exec.(*Evaluator).evalMethod(0x14000112c80, {0x1043a8118, 0x14000056298}, {0x1400015c116, 0xa}, {0x14000010ec0, 0x1, 0x1?}, 0x14000201530)
        /Users/A108073420/go/pkg/mod/github.com/nikolalohinski/gonja/v2@v2.3.1/exec/calls.go:113 +0x658
github.com/nikolalohinski/gonja/v2/exec.(*Evaluator).evalCall(0x14000112c80, 0x1400013c460)
        /Users/A108073420/go/pkg/mod/github.com/nikolalohinski/gonja/v2@v2.3.1/exec/calls.go:23 +0x5d4
github.com/nikolalohinski/gonja/v2/exec.(*Evaluator).Eval(0x0?, {0x1043a7f88?, 0x1400013c460?})
        /Users/A108073420/go/pkg/mod/github.com/nikolalohinski/gonja/v2@v2.3.1/exec/evaluator.go:52 +0x62c
github.com/nikolalohinski/gonja/v2/exec.(*Renderer).Eval(0x14000212300, {0x1043a7f88, 0x1400013c460})
        /Users/A108073420/go/pkg/mod/github.com/nikolalohinski/gonja/v2@v2.3.1/exec/renderer.go:149 +0xa8
github.com/nikolalohinski/gonja/v2/builtins/control_structures.(*IfControlStructure).Execute(0x14000101740, 0x14000212300, 0x1400015c106?)
        /Users/A108073420/go/pkg/mod/github.com/nikolalohinski/gonja/v2@v2.3.1/builtins/control_structures/if.go:30 +0x74
github.com/nikolalohinski/gonja/v2/exec.(*Renderer).Visit(0x14000212300, {0x1043a7dd0?, 0x1400020d410})
        /Users/A108073420/go/pkg/mod/github.com/nikolalohinski/gonja/v2@v2.3.1/exec/renderer.go:114 +0x668
github.com/nikolalohinski/gonja/v2/nodes.Walk({0x1043a6f88?, 0x14000212300?}, {0x1043a7dd0, 0x1400020d410})
        /Users/A108073420/go/pkg/mod/github.com/nikolalohinski/gonja/v2@v2.3.1/nodes/walk.go:12 +0x3c
github.com/nikolalohinski/gonja/v2/nodes.Walk({0x1043a6f88?, 0x14000212300?}, {0x1043a7d30, 0x140001010c0})
        /Users/A108073420/go/pkg/mod/github.com/nikolalohinski/gonja/v2@v2.3.1/nodes/walk.go:29 +0x154
github.com/nikolalohinski/gonja/v2/exec.(*Renderer).ExecuteWrapper(0x140002121c0?, 0x140001010c0)
        /Users/A108073420/go/pkg/mod/github.com/nikolalohinski/gonja/v2@v2.3.1/exec/renderer.go:126 +0x3c
github.com/nikolalohinski/gonja/v2/builtins/control_structures.(*ForControlStructure).Execute(0x1400013c370, 0x140002121c0, 0x1400015c0e6?)
        /Users/A108073420/go/pkg/mod/github.com/nikolalohinski/gonja/v2@v2.3.1/builtins/control_structures/for.go:158 +0x55c
github.com/nikolalohinski/gonja/v2/exec.(*Renderer).Visit(0x140002121c0, {0x1043a7dd0?, 0x1400020d440})
        /Users/A108073420/go/pkg/mod/github.com/nikolalohinski/gonja/v2@v2.3.1/exec/renderer.go:114 +0x668
github.com/nikolalohinski/gonja/v2/nodes.Walk({0x1043a6f88?, 0x140002121c0?}, {0x1043a7dd0, 0x1400020d440})
        /Users/A108073420/go/pkg/mod/github.com/nikolalohinski/gonja/v2@v2.3.1/nodes/walk.go:12 +0x3c
github.com/nikolalohinski/gonja/v2/nodes.Walk({0x1043a6f88?, 0x140002121c0?}, {0x1043a7d08, 0x14000098080})
        /Users/A108073420/go/pkg/mod/github.com/nikolalohinski/gonja/v2@v2.3.1/nodes/walk.go:23 +0x1ac
github.com/nikolalohinski/gonja/v2/exec.(*Renderer).Execute(...)
        /Users/A108073420/go/pkg/mod/github.com/nikolalohinski/gonja/v2@v2.3.1/exec/renderer.go:136
github.com/nikolalohinski/gonja/v2/exec.(*Template).Execute(0x14000096050, {0x1043a6e28, 0x14000056020}, 0x4?)
        /Users/A108073420/go/pkg/mod/github.com/nikolalohinski/gonja/v2@v2.3.1/exec/template.go:73 +0x1c8
main.main()
```

--- a/exec/renderer.go
+++ b/exec/renderer.go
@@ -44,6 +44,7 @@ func (r *Renderer) Inherit() *Renderer {
 			Tests:             r.Environment.Tests,
 			Filters:           r.Environment.Filters,
 			ControlStructures: r.Environment.ControlStructures,
+			Methods:           r.Environment.Methods,
 		},
 		Template: r.Template,
 		RootNode: r.RootNode,

--- a/tests/integration/strings_test.go
+++ b/tests/integration/strings_test.go
@@ -194,5 +194,33 @@ var _ = Context("strings", func() {
 				AssertPrettyDiff("one two", *returnedResult)
 			})
 		})
+		Context("https://github.com/NikolaLohinski/gonja/issues/25", func() {
+			BeforeEach(func() {
+				*loader = loaders.MustNewMemoryLoader(map[string]string{
+					*identifier: heredoc.Doc(`
+					{%- set ns = namespace(found=false) -%}
+					{%- for v in ["a", "b", "c"] -%}
+					      {%- if v.startswith("b") -%}
+					        {% set ns.found=true -%}
+					      {%- endif -%}
+					{%- endfor -%}
+					{{ ns.found }}
+					{{ name }}
+					`),
+				})
+				(*environment).Context.Set("name", "bob")
+			})
+
+			It("should return the expected rendered content", func() {
+				By("not returning any error")
+				Expect(*returnedErr).To(BeNil())
+				By("returning the expected result")
+				expected := heredoc.Doc(`
+				True
+				bob
+				`)
+				AssertPrettyDiff(expected, *returnedResult)
+			})
+		})
 	})
 })

